### PR TITLE
Alation crawler

### PR DESCRIPTION
**Pull request by usefulalgorithm**
### 🤔 Why?

To allow migrating from Alation to Metaphor, a crawler that extracts user defined information from Alation is necessary.

### 🤓 What?

Implemented an Alation crawler that extracts the following:
- table / column descriptions
- table / column tags
- table steward as owner

In Alation the concept of data ownership does not exist. The closest thing we have is Steward.

### 🧪 Tested?

Did unit test with everything mocked.

Unable to test with real Alation instance, don't have access to one.

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

File: metaphor/alation/client.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import requests
+
+
+class Client:
+    def __init__(self, base_url: str, headers: Dict[str, str]) -> None:
+        self.base_url = base_url
+        self.headers = headers
+
+    def get(self, path: str, params: Optional[Dict[str, Any]] = None):
```
Comment (elic-eon) [inline metaphor/alation/client.py:11]: add return type?

File: metaphor/alation/schema.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            return DataPlatform.MYSQL
+        if self.dbtype == "postgresql":
+            return DataPlatform.POSTGRESQL
+        if self.dbtype == "sqlserver":
+            return DataPlatform.MSSQL
+        if self.dbtype == "redshift":
+            return DataPlatform.REDSHIFT
+        if self.dbtype == "snowflake":
+            return DataPlatform.SNOWFLAKE
+        return DataPlatform.EXTERNAL
```
Comment (elic-eon) [inline metaphor/alation/schema.py:100]: Do we want to capture platforms that we do not currently support as external?

Comment (usefulalgorithm) [reply metaphor/alation/schema.py:115]: It's either `UNKNOWN`, `EXTERNAL` or `None`. I think FE picks up none of those though

Comment (elic-eon) [reply metaphor/alation/schema.py:115]: In this case, shouldn't we just skip datasets we have not supported yet?

Comment (usefulalgorithm) [reply metaphor/alation/schema.py:115]: good call, will do that

File: metaphor/alation/extractor.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+            ownership_assignment=self._get_ownership_assignment(table.steward),
+        )
+        return dataset
+
+    async def extract(self) -> Collection[ENTITY_TYPES]:
+        for table_obj in self._client.get("integration/v2/table/", {"limit": 100}):
+            try:
+                self._entities.append(self._init_dataset(table_obj))
+            except Exception:
+                logger.exception(f"Cannot extract table, obj = {table_obj}")
```
Comment (elic-eon) [inline metaphor/alation/extractor.py:160]: Log the error message as well? And do we capture the exception types we want here?

**Comment by elic-eon** [comment]
@mars-lan Good to go?

**Comment by mars-lan** [comment]
> @mars-lan Good to go?

Let's also map stewards to ownership?

**Comment by usefulalgorithm** [comment]
> > @mars-lan Good to go?
> 
> Let's also map stewards to ownership?

I did that in this PR

**Comment by mars-lan** [comment]
> > > @mars-lan Good to go?
> > 
> > 
> > Let's also map stewards to ownership?
> 
> I did that in this PR: https://github.com/MetaphorData/connectors/pull/761/files#diff-7488362cd299580dfc58558741f3c458d8f336fa8a38742c3335c741d468f06fR108

Got it. Please update PR description, thanks.

**Review by mars-lan — APPROVED** [review]
Otherwise LGTM

**Review by mars-lan — APPROVED** [review]

**Pull request merged**